lib/ts/recipe/emailverification/emailVerificationAuth.tsx

```diff
@@ -29,45 +29,44 @@ const EmailVerificationAuth: React.FC<Props> = ({ children, ...props }) => {
 
     const [isEmailVerified, setIsEmailVerified] = useState(false);
 
-    // we extract these three this way so that the useEffect below
-    // doesn't rerun just because the sessionContext or props objects
-    // have changed, even though the doesSessionExist & emailVerificationMode
-    // have not.
-    const doesSessionExist = sessionContext.loading === false && sessionContext.doesSessionExist;
     const emailVerificationMode = props.recipe.config.mode;
     const propsRef = React.useRef(props);
     const userContext = useUserContext();
 
     const checkIsEmailVerified = useCallback(async () => {
-        if (doesSessionExist && emailVerificationMode === "REQUIRED") {
+        if (sessionContext.loading === true) {
+            // This callback should only be called if the session is already loaded
+            throw new Error("Should never come here");
+        }
+        if (sessionContext.doesSessionExist && emailVerificationMode === "REQUIRED") {
             return (await propsRef.current.recipe.isEmailVerified(userContext)).isVerified;
         }
         return undefined;
-    }, [doesSessionExist, emailVerificationMode]);
+    }, [sessionContext, emailVerificationMode]);
     const useIsEmailVerified = useCallback(
         async (isEmailVerified: boolean | undefined) => {
-            if (doesSessionExist && emailVerificationMode === "REQUIRED") {
+            if (sessionContext.loading === true) {
+                // This callback should only be called if the session is already loaded
+                throw new Error("Should never come here");
+            }
+            if (sessionContext.doesSessionExist && emailVerificationMode === "REQUIRED") {
                 if (isEmailVerified === false) {
                     await propsRef.current.recipe.redirect({ action: "VERIFY_EMAIL" }, propsRef.current.history);
                 } else {
                     setIsEmailVerified(true);
                 }
             }
         },
-        [doesSessionExist, emailVerificationMode]
+        [sessionContext.loading, emailVerificationMode]
     );
 
     useOnMountAPICall(checkIsEmailVerified, useIsEmailVerified, undefined, sessionContext.loading === false);
 
-    if (doesSessionExist === false) {
-        return <>{children}</>;
-    }
```

Comment:
this should not be removed IMO cause if requireAuth is false, and a session does not exist, it should show the child

Reply:
Maybe add a test about this